pkg/scheduler/framework/interface.go

```diff
@@ -50,9 +50,121 @@ type NodeScore struct {
 	Score int64
 }
 
-// NodeToStatusMap contains the statuses of the Nodes where the incoming Pod was not schedulable.
-// A PostFilter plugin that uses this map should interpret absent Nodes as UnschedulableAndUnresolvable.
-type NodeToStatusMap map[string]*Status
+// NodeToStatusReader is a read-only interface of NodeToStatus passed to each PostFilter plugin.
+type NodeToStatusReader interface {
+	// Get returns the status for given nodeName.
+	// If the node is not in the map, the AbsentNodesStatus is returned.
+	Get(nodeName string) *Status
+	// NodesForStatusCode returns a list of NodeInfos for the nodes that have a given status code.
+	// It returns the NodeInfos for all matching nodes denoted by AbsentNodesStatus as well.
```

Comment:
nit: clarify it fetches all Nodes first and then filter them out based on status

```suggestion
	// NodesForStatusCode fetches all existing Nodes and returns the nodes that either explicitly have a given status code, 
	// or nodes that don't have a status explicitly when absentNodesStatus matches a given status code.
```

Reply:
Moved fetching all nodes to the branch where `absentNodesStatus` is matching given code, so altered the comment differently.